planner/optimize.go

```diff
@@ -100,6 +100,26 @@ func GetExecuteForUpdateReadIS(node ast.Node, sctx sessionctx.Context) infoschem
 // The node must be prepared first.
 func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (plannercore.Plan, types.NameSlice, error) {
 	sessVars := sctx.GetSessionVars()
+	useBinding := sessVars.UsePlanBaselines
+	stmtNode, ok := node.(ast.StmtNode)
+	if !ok {
+		useBinding = false
+	}
+	var (
+		bindRecord *bindinfo.BindRecord
+		scope      string
+		err        error
+	)
+	if useBinding {
+		bindRecord, scope, err = getBindRecord(sctx, stmtNode) // get the corresponding binding record before adding the extra limit
```

Comment:
Function `TryFastPlan` is executed behind function `getBindRecord`. So some `PointGet/ BatchPointGet` plans will be changed. This should be noticed.